src/Sarif.Viewer.VisualStudio/OpenLogFileCommands.cs

```diff
@@ -168,117 +165,84 @@ private async System.Threading.Tasks.Task MenuItemCallbackAsync(object sender, E
-                    {
-                        toolFormat = ToolFormat.FortifyFpr;
-                        title = "Open Fortify FPR log file";
-                        filter = "Fortify FPR log files (*.fpr)|*.fpr";
-                        break;
-                    }
-                    case OpenCppCheckFileCommandId:
-                    {
-                        toolFormat = ToolFormat.CppCheck;
-                        title = "Open CppCheck XML log file";
-                        filter = "CppCheck log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenClangFileCommandId:
-                    {
-                        toolFormat = ToolFormat.ClangAnalyzer;
-                        title = "Open Clang XML log file";
-                        filter = "Clang log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenAndroidStudioFileCommandId:
-                    {
-                        toolFormat = ToolFormat.AndroidStudio;
-                        title = "Open Android Studio XML log file";
-                        filter = "Android Studio log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenSemmleFileCommandId:
-                    {
-                        toolFormat = ToolFormat.SemmleQL;
-                        title = "Open Semmle QL CSV log file";
-                        filter = "Semmle QL log files (*.csv)|*.csv";
-                        break;
-                    }
-                    case OpenPylintFileCommand:
-                    {
-                        toolFormat = ToolFormat.Pylint;
-                        title = "Open Pylint JSON log file";
-                        filter = "Pylint log files (*.json)|*.json";
-                        break;
-                    }
-                    case OpenTSLintFileCommand:
-                    {
-                        toolFormat = ToolFormat.TSLint;
-                        title = "Open TSLint JSON log file";
-                        filter = "TSLint log files (*.json)|*.json";
-                        break;
-                    }
+                    string resourceName = string.Format(CultureInfo.InvariantCulture, "{0}{1}{2}", FilterResourceNamePrefix, fieldInfo.Name, FilterResourceNameSuffix);
+                    string openFileDialogFilterString = Resources.ResourceManager.GetString(resourceName, CultureInfo.CurrentCulture);
+                    fieldInfoToOpenFileDialogFilterDisplayString.Add(new KeyValuePair<FieldInfo, string>(fieldInfo, openFileDialogFilterString));
                 }
 
-                filter += "|All files (*.*)|*.*";
+                // Sort the filters by their display strings so the user has a nice alphabetized list.
+                IEnumerable<KeyValuePair<FieldInfo, string>> orderedFilters = fieldInfoToOpenFileDialogFilterDisplayString.OrderBy(kvp => kvp.Value);
```

Comment:
>Value [](start = 142, length = 5)

The problem is that now SARIF will sort alphabetically. Before, when it was treated specially, SARIF had pride of place at the top of the list.

I won't hold the PR for this but I think it should be fixed.